Deploy/stacks/db/fileserver/fileserver/src/main/java/com/cmclinnovations/fileserver/FileServer.java

```diff
@@ -0,0 +1,198 @@
+@MultipartConfig(
+fileSizeThreshold = FileServer.ONE_MB_IN_B,
+location = "/app/fs_root/",
+maxFileSize = FileServer.ONE_HUNDRED_MB_IN_B,
+maxRequestSize = FileServer.ONE_GB_IN_B
+)
+public class FileServer extends HttpServlet {
+
+    // URL Patterns
+    static final String DELETE_URL_PATTERN = "/delete/*";
+    static final String DOWNLOAD_URL_PATTERN = "/download/*";
+    static final String UPLOAD_URL_PATTERN = "/upload";
+
+    // Some constants to set size limits
+    static final int ONE_MB_IN_B = 1024 * 1024;
+    static final int ONE_HUNDRED_MB_IN_B = 100 * ONE_MB_IN_B;
+    static final int ONE_GB_IN_B = 1000 * ONE_MB_IN_B;
+
+    // Content type prefixes
+    private static final String MULTIPART_FORM_TYPE_PREFIX = "multipart/form-data";
+    private static final String MULTIPART_MIXED_TYPE_PREFIX = "multipart/mixed";
+
+    private static final Logger LOGGER = LogManager.getLogger(FileServer.class);
+
+    private synchronized String writeFilePart(Part filePart, String subDirStr_in) {
+        String fNameRequested = filePart.getSubmittedFileName();
+        String subDirStr = subDirStr_in == null ? "." : subDirStr_in;
+
+        String fName = fNameRequested;
+        Path destPath = Paths.get(subDirStr).resolve(fName);
+
+        // If requested filename exists, append a suffix generated from the current sytem time
+        if (Files.exists(destPath)) {
+            String fNameBase = FilenameUtils.getBaseName(fNameRequested);
+            String ext = FilenameUtils.getExtension(fNameRequested);
+            String suffix = Long.toString(System.currentTimeMillis());
+
+            // Append suffix, allowing for paths with no extension
+            fName = (ext.length() > 0) ? fNameBase + "_" + suffix + "." + ext : fNameBase + suffix;
+            destPath = Paths.get(subDirStr).resolve(fName);
+        }
+
+        try {
+            // Create the subdirectory, if necessary
+            File subDir = new File(subDirStr);
+            if (!subDir.exists()) {
+                //LOGGER.info("Creating subdirectory " + subDirStr);
+                boolean dirCreated = subDir.mkdirs();
+                if (!dirCreated) {
+                    LOGGER.error("Failed to created sub-directory " + subDirStr);
+                    return null;
+                }
+            }
+
+            // Write the file
+            filePart.write(destPath.toString());
+        } catch (IOException ex) {
+            LOGGER.error("writeFilePart: IOException when trying to write file " + destPath, ex);
+            return null;
+        }
+
+        /*
+         * Wait for a millisecond to ensure the next call of this function never generates the same
+         * fallback suffix.
+         */
+        try {
+            wait(1);
+        } catch (InterruptedException ex) {
+            LOGGER.error("writeFilePart: (Thread)InterruptedException whilst pausing at the end of the method", ex);
+        }
+        return destPath.toString();

```

Comment:
Return the full file path/URL so that it can be used as an argument of the get or delete methods. See issue #127.